Updates (#1)
fatlama/backend-challenge: README.md

# Backend Challenge

## Introduction

> Fat Llama relies heavily on our search in order for users to be able to find the items they need from other users. Our search must return a list of items matching a given search term combined with each item owners information.

Reviewer: ```suggestion
Fat Llama relies heavily on our search in order for users to be able to find the items they need from other users. Our search must return a list of items matching a given search term, as well as displaying certain information about the item owners (see example Item Card below).
```

On the production web & mobile app there are other factors that come into play such as lender rating, response time, categories, time since listing, and more. For this challenge though, we want you to focus only on the above.

## The Challenge

> We want you to build a `GET /search` endpoint that will return the most appropriate 20 items given a `searchTerm`, e.g. `/search?searchTerm=camera`. The items returned must contain the owners information so we can correctly display them in the Fat Llama frontend. You are not required to incorporate any complex fuzzy text matching.

Reviewer: ```suggestion
We want you to build a `GET /search` endpoint that will return the most appropriate 20 items given a `searchTerm`, e.g. `/search?searchTerm=camera`. The items returned must contain the owner's information so we can correctly display them in the Fat Llama frontend. You are not required to incorporate any complex fuzzy text matching.
```

The schema for the items returned should be:

```javascript
{
  itemId: number,
  itemName: string,
  itemPrice: number,
  userFirstName: string,
  userCity: string
}

> ```

Reviewer: I think maybe we should just have a picture of an item card with this info on, and leave them to come up with the schema. E.g. it might be nicer to structure it as:
```typescript
interface User {
  userFirstName: string
  userCity: string
}
interface ItemWithUser {
  itemId: number
  itemName: string
  itemPrice: number
  user: User
}
interface ApiResponse {
  page: number,
  results: ItemWithUser[]
}
```

> You are provided with two csvs containing approximately 1000 items and 300 users. Its is up you how your API handles these.

Reviewer 2: ```suggestion
You are provided with two csvs containing approximately 1000 items and 300 users respectively. It's up to you how your API handles these.
```

Reviewer: "Respectively" is redundant here

Reviewer: "Respectively" is redundant here

When you are finished, write up a short summary of why you made the choices you did in terms of technology and design. This should be no more than 500 words.

## Things to think about:

- Think about points of failure and how your endpoint will perform under load.

> - Language/frameworks: nodeJs / Typescript is preferrable, native javascript is also acceptable.

Reviewer: ```suggestion
- Language/frameworks: Node.js with Typescript is preferable, but if you haven't worked with Typescript, Node.js + Javascript is also acceptable.
```
